packages/app/lib/common/index.js

```diff
@@ -161,9 +246,98 @@ export function createMessage(moduleName, methodName, uniqueMessage = '') {
   }
 }
 
+function getNamespace(target) {
+  if (target.GeoPoint) {
+    // target is statics object. GeoPoint is a static class on Firestore
+    return 'firestore';
+  }
+  if (target._config && target._config.namespace) {
+    return target._config.namespace;
+  }
+  const className = target.name ? target.name : target.constructor.name;
+  return Object.keys(mapOfDeprecationReplacements).find(key => {
+    if (mapOfDeprecationReplacements[key][className]) {
+      return key;
+    }
+  });
+}
+
+function getInstanceName(target) {
+  if (target.GeoPoint) {
+    // target is statics object. GeoPoint is a static class on Firestore
+    return 'statics';
+  }
+  if (target._config) {
+    // module class instance, we use default to store map of deprecated methods
+    return 'default';
+  }
+  if (target.name) {
+    // It's a function which has a name property unlike classes
+    return target.name;
+  }
+  // It's a class instance
+  return target.constructor.name;
+}
+
+export function createDeprecationProxy(instance) {
```

Comment:
This got a bit trickier than anticipated because of the nature of Firestore statics. Some are classes that only have static properties, Timestamp is a class with static methods and can be instantiated, setLogLevel() is a function, Filter is a function and also has static methods 😓 .

Reply:
Years of warts have grown all over this thing 😆 , I can take credit for setLogLevel, sorry about that

Reply:
ha, no worries, Mike. I'm guilty for Filter!